src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -948,4 +983,173 @@ public static boolean isInSparseFormat(MatrixObject mo) {
 			return mo.getGPUObject().isInSparseFormat();
 		return MatrixBlock.evalSparseFormatInMemory(mo.getNumRows(), mo.getNumColumns(), mo.getNnz());
 	}
+	
+	/**
+	 * Performs elementwise multiplication of input vector in and constant using cublasDaxpy
+	 * 
+	 * @param ec
+	 * @param in
+	 * @param constant
+	 * @param outputName
+	 * @param op
+	 * @throws DMLRuntimeException
+	 */
+	public static void vectorScalarMult(ExecutionContext ec, MatrixObject in, double constant, String outputName, GPUEnabledElementwiseOp op) throws DMLRuntimeException {
```

Comment:
rename vectorScalarMult -> vectorMatrixElementWise
Also you might want to stick with naming convention in matrix consistent with [cuBlasDaxpy](http://docs.nvidia.com/cuda/cublas/#cublas-lt-t-gt-axpy) since this function is a wrapper to that.

Also, this function seems to be take in a "control parameter", namely the "op".
My preference matches this [guy](http://programmers.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior), but then again, this is a matter of design and is subjective. Also, since an element wise divide can be done, the name of the function does not make sense anymore.
I would split it up into 2 functions - _vectorScalarElementWiseMult_ & _vectorScalarElementWiseDiv_